crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs

```diff
@@ -263,3 +263,73 @@ void __swift_bridge__$some_function(struct __private__ResultPtrAndPtr arg);
         .test();
     }
 }
+
+/// Test code generation for Rust function that accepts a Result<T, E> where T is (), and E is an
```

Comment:
Mind adding one more test module that tests `Result<struct UnitStruct;, OpaqueRust>`?

This way we can be confident that the suggestions above work.